va_explorer/va_data_management/views.py

```diff
@@ -1,96 +1,92 @@
-# Revert the most recent change
-# TODO: Should record automatically be recoded?
-@login_required
-def revert_latest(request, id):
-    va = VerbalAutopsy.objects.get(id=id)
-    if va.history.count() > 1:
-        previous = va.history.all()[1]
-        latest = va.history.latest()
-        if len(latest.diff_against(previous).changes) > 0:
-            previous.instance.save()
-        return redirect("va_data_management:show", id=id)
+from va_explorer.va_data_management.models import VerbalAutopsy, CauseOfDeath, CauseCodingIssue, Location
+from va_explorer.va_data_management.forms import VerbalAutopsyForm
+from va_explorer.utils.mixins import CustomAuthMixin
+
+
+class Index(CustomAuthMixin, ListView):
+    template_name = 'va_data_management/index.html'
+    paginate_by = 15
+    queryset = VerbalAutopsy.objects.prefetch_related("location", "causes", "coding_issues").order_by("id")
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['object_list'] = [{
+            "id": va.id,
+            "name": va.Id10007,
+            "facility": va.location.name,
+            "cause": va.causes.all()[0].cause if len(va.causes.all()) > 0 else "",
+            "warnings": len([issue for issue in va.coding_issues.all() if issue.severity == 'warning']),
+            "errors": len([issue for issue in va.coding_issues.all() if issue.severity == 'error'])
+        } for va in context['object_list']]
+
+        return context
+
+
+class Show(CustomAuthMixin, DetailView):
+    template_name = 'va_data_management/show.html'
+    model = VerbalAutopsy
+    pk_url_kwarg = 'id'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['form'] = VerbalAutopsyForm(None, instance=self.object)
+
+        coding_issues = self.object.coding_issues.all()
+        context['warnings'] = [issue for issue in coding_issues if issue.severity == 'warning']
+        context['errors'] = [issue for issue in coding_issues if issue.severity == 'error']
+
+        # TODO: date in diff info should be formatted in local time
+        # TODO: history should eventually show user who made the change
+        history = self.object.history.all().reverse()
+        history_pairs = zip(history, history[1:])
+        context['diffs'] = [new.diff_against(old) for (old, new) in history_pairs]
+
+        return context
+
+
+class Edit(CustomAuthMixin, UpdateView):
+    template_name = 'va_data_management/edit.html'
+    form_class = VerbalAutopsyForm
+    model = VerbalAutopsy
+    pk_url_kwarg = 'id'
+
+    def get_success_url(self):
+        return reverse('va_data_management:show', kwargs={'id': self.object.id})
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['id'] = self.object.id
+        return context
+
+
+class Reset(CustomAuthMixin, DetailView):
+    model = VerbalAutopsy
+    pk_url_kwarg = 'id'
+
```

Comment:
Same comment here

import SuccessMessageMixin, add something like:

success_message = "Verbal Autopsy changes successfully reverted!"

Reply:
Done. Couldn't use SuccessMessageMixin here because it's not a form view, but I just imported messages and set the success message directly. Same effect.